ext/Recorder.php

```diff
@@ -617,19 +630,18 @@ protected function isStepIgnored(StepEvent $e)
     }
 
     /**
-     * @param StepEvent|TestEvent $e
-     *
+     * @param StepEvent|TestEvent $event
      * @return string
      */
-    private function getTestName($e)
+    private function getTestName($event): string
     {
-        return basename($e->getTest()->getMetadata()->getFilename()) . '_' . preg_replace('/[^A-Za-z0-9\-\_]/', '_', $e->getTest()->getMetadata()->getName());
+        return basename($event->getTest()->getMetadata()->getFilename()) . '_' . preg_replace('/[^A-Za-z0-9\-\_]/', '_', $event->getTest()->getMetadata()->getName());
     }
 
     /**
      * @param string $message
      */
-    protected function writeln($message)
+    protected function writeln($message): void
```

Comment:
Parameter is a string

Reply:
Abstract class `src/Codeception/Extension.php` needs to be modified, I will do it in a future PR.

Reply:
I'm waiting for https://github.com/Codeception/Codeception/pull/6071 to be merged so as not to generate conflicts.

Reply:
Then take a break, we have 2 months.

Reply:
At the moment I think I can move forward in `src/PHPUnit` but pending folders or files in `src/Codeception` are being occupied by **DavertMik**.

Reply:
Of course, I will only have one PR of this type open at a time, only when you merge one will I continue with another.

Reply:
Then don't worry about conflicts, I will rebase other branches after merging this one.

Majority of code in src/PHPUnit was copied from PHPUnit. It would be the best to keep differences from original minimal and refresh them with changes implemented by PHPUnit.
Please update files in src/PHPUnit/Constraint/ directory, /src/PHPUnit/Util/Log/ directory, ConsolePrinter and FilterTest only.

Reply:
I understood, I will, thanks

Reply:
I also fixed this